tv-shows/src/components/features/shows/ShowPicker/components/ShowPickerStep.tsx

```diff
@@ -0,0 +1,42 @@
+'use client';
+
+import { Flex } from '@chakra-ui/react';
+import { useContext } from 'react';
+import { IShow } from '../../../../../typings/Show.type';
+import { ShowCard } from '../../../../shared/ShowCard/ShowCard';
+import { ShowPickerContext } from './ShowPickerContextProvider';
+
+export const ShowPickerStep = () => {
+	const { activeStep, availableShows, setSelectedShows, selectedShows } = useContext(ShowPickerContext);
+	if (!availableShows) return null;
+
+	const activeStepShows = availableShows.slice(activeStep * 4, (activeStep + 1) * 4);
+
+	const toggleShowSelection = (show: IShow) => {
+		const isSelected = selectedShows.some((selectedShow) => selectedShow.id === show.id);
+
+		if (isSelected) {
+			setSelectedShows(selectedShows.filter((selectedShow) => selectedShow.id !== show.id));
+		} else {
+			const filteredShows = selectedShows.filter(
+				(selectedShow) => !activeStepShows.some((stepShow) => stepShow.id === selectedShow.id)
+			);
+
+			setSelectedShows([...filteredShows, show]);
+		}
+	};
+
+	return (
+		<Flex flexDirection={{ base: 'column', md: 'row' }} align="center" justify="space-between" gap="8">
+			{activeStepShows.map((availableShow) => (
+				<ShowCard
```

Comment:
![image](https://github.com/user-attachments/assets/6cb9da0b-d4e7-45e8-8cda-f9206a236e87)
What kind of poopy UI is this?

Reply:
Discussed this on consultations